crates/apollo-compiler/src/diagnostics.rs

```diff
@@ -410,12 +411,12 @@ pub struct UnsupportedLocation {
     pub ty: String,
 
     // current location where the directive is used
-    pub dir_loc: String,
+    pub dir_loc: DirectiveLocation,
 
     #[source_code]
     pub src: Arc<str>,
 
-    #[label("{} is not a valid location", self.dir_loc)]
+    #[label("{} is not a valid location for this directive", self.dir_loc)]
```

Comment:
didn't catch this in review earlier but I think this message could be improved a bit … maybe something like "cannot use directive `@xyz` here"

Reply:
I guess the thing is that the label is more visually prominent than the "main" error message, despite our "main" message being the thing that contains the important information
![image](https://user-images.githubusercontent.com/1006268/213755145-3164ce6e-d86a-4d0a-a7f4-5b44ce650ec6.png)

Reply:
oh yea this is nicer ty!